render/marquee.go

```diff
@@ -102,35 +124,140 @@ func (m Marquee) Paint(bounds image.Rectangle, frameIdx int) image.Image {
 		offend = -cw
 	}
 
+	var offset int
+	if m.isAlternating() {
+		offset = m.AlternatingOffset(cw, size, frameIdx)
+	} else {
+		offset = m.ScrollingOffset(cw, size, frameIdx)
+	}
+
+	var dc *gg.Context
+	if m.isVertical() {
+		dc = gg.NewContext(im.Bounds().Dx(), m.Height)
+		dc.DrawImage(im, 0, offset)
+	} else {
+		dc = gg.NewContext(m.Width, im.Bounds().Dy())
+		dc.DrawImage(im, offset, 0)
+	}
+
+	return dc.Image()
+}
+
+func (m Marquee) ScrollingFrameCount(cw int, size int) int {
+	offstart := m.OffsetStart
+	if offstart < -cw {
+		offstart = -cw
+	}
+
+	offend := m.OffsetEnd
+	if offend < -cw {
+		offend = -cw
+	}
+
+	return cw + offstart + size - offend + 1
```

Comment:
I did not change this, but I actually believe this additional `1` frame is a mistake. If you look at the last example image I posted in this PR, after the 1 frame is repeated at offset 0 and it seems like it "lags".

Reply:
I think you may be correct. Can't figure out why it's there.